### PR TITLE
Fixes after selection preview rework

### PR DESCRIPTION
**Pull request by kubecz3k**
Selected shape preview behaves as it appears inconsistently

-----------
    fix selection-preview for flat theme 🔨
    
    Flat theme has never been updated to work in new selection preview
    This commit fixes this oversight by applying changes needed for it to
    work

    refresh selection preview for shapes and motors 🔧
    
    This commit ensures the selection previews for shapes and motors will
    update on every motor/shape transform change

### CODE REVIEW

**Comment by kubecz3k** [comment]
I have seen this pr is needed in order to unblock a task, please feel free to merge it as soon as it's approved. 
I will address any eventual feedback in subsequent pr

**Comment by MarianoGnu** [comment]
> I have seen this pr is needed in order to unblock a task, please feel free to merge it as soon as it's approved. I will address any eventual feedback in subsequent pr

That task is blocked only because the regression of this PR was wrongly attributed to the PR of that task, there's no rush because this will fix it

File: godot_project/project_workspace/workspace_context/workspace_context.gd
Comment (MarianoGnu) [inline godot_project/project_workspace/workspace_context/workspace_context.gd:11]: FYI: #54 intriduced [`signal virtual_object_transform_changed(structure_context: StructureContext)`](https://github.com/MSEP-one/msep.one/blob/a85d4ba3cdc11a4775a989cad43ef29ffb837444/godot_project/project_workspace/workspace_context/workspace_context.gd#L11)

**Review by MarianoGnu — CHANGES_REQUESTED** [review]
Code looks good, but please rebase the code to make use of the already existing `virtual_object_transform_changed` signal, which as a plus will also consider `NanoVirtualAnchor`, not covered in this PR

**Review by MarianoGnu — APPROVED** [review]

**Pull request merged**
